File: 198 House Robber/solution.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def rob(self, nums):
        odd,even = 0,0
        for i in range(len(nums)):
            if i%2==0:
                even = max(odd,even+nums[i])
            else:
                odd = max(even,odd+nums[i])
        return max(even,odd)

File: 198 House Robber/solution.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int odd=0;
        int even=0;
        for (int i=0;i<nums.size();i++){
            if (i%2)
                odd = max(odd+nums[i],even);
            else
                even = max(odd,even+nums[i]);
        }
        return max(odd,even);
    }
};